server/scripts/commands_main.py

```diff
@@ -29,26 +29,21 @@ def convert():
 
 @convert.command(name="kpf2dive")
 @click.argument('inputs', type=click.File('rb'), nargs=-1)
-@click.option('--output', type=click.File('wt'), default='result.json')
-def convert_kpf(inputs: List[BinaryIO], output: TextIO):
-    def read_in_chunks(file_object: BinaryIO, chunk_size=524288):
-        """
-        Lazy function (generator) to read a file piece by piece.
-        """
-        while True:
-            data = file_object.read(chunk_size)
-            if not data:
-                break
-            yield data
-
-    tracks = meva.load_kpf_as_tracks([read_in_chunks(file) for file in inputs])
-    json.dump(tracks, output)
+@click.option('--output', type=click.File('wt'), default='annotations.dive.json')
+@click.option('--output-attrs', type=click.File('wt'), default='attributes.json')
+def convert_kpf(inputs: List[BinaryIO], output: TextIO, output_attrs: TextIO):
+    lines = [line.decode() for file in inputs for line in list(file.read().splitlines())]
+    input = '\n'.join(lines)
+    annotations, attributes = kpf.convert(kpf.load(input))
```

Comment:
This is really picky of me so feel free to ignore.  The JSON files from zip exporting give a sorted Dict where the keys are sorted ascending.  The Endpoint for annotations gives an array of track objects that are also sorted.
Could we  sort the exports from these commands, it would make it a bit easier to do comparisons between the zip download and the command line conversion.  I know that the order doesn't matter, but the comparison process is what let me to find the 'version' key missing.

something like just passing the output of group/tracks through `dict(sorted(annotations['tracks'].items()))`

Reply:
fixed